### PR TITLE
Deleting all images that have passed tests before upload

### LINKED ISSUE #23400: Only upload failed images on failure
Currently when a job fails on GHA we upload all 109 MB across ~7k files which takes a surprisingly long time.  What we really want is just the images that failed and the computed difference.

This is labeled as "good first issue" because there are no API designs here (it is all configuring CI).

Steps:

 - sort out how to identify the failed test images (there is a systematic naming convention)
 -  explain that to the uploader workflow (or just delete all of the non-failed fails!)

### PR DESCRIPTION
**Pull request by Impaler343**
## PR summary

Supersedes PR #27881 (I'm so sorry for creating unnecessary traffic on this repo.)
Attempts to close #23400 and potentially complete what PR #19466 intended to do
Included a bash script right before the uploading part of the code in the *tests.yml* file to delete all files(.png, .svg, .eps, .pdf) that have not failed in the *result_images* directory. Have tested this using a dummy bash file and the downloaded result_images folder. Please let me know if this approach seems correct, or is blatantly wrong :).

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by Impaler343** [comment]
Hey, could someone help me out on how to resolve the failing tests?

**Comment by rcomer** [comment]
@Impaler343 those tests seem to be a bit flakey at the moment as they are failing across multiple PRs.  I have re-spun them so we can see if they pass this time.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -318,6 +318,35 @@ jobs:
...
+                  fi
+                  if [[ ! -e "${base}-failed-diff.$extension" ]]; then
+                      if [[ -e "$file" ]]; then
+                          rm "$file"
+                          echo "Removed $file"
+                      fi
+                      if [[ -e "${base}.$extension" ]]; then
+                          rm "${base}.$extension"
+                      fi
+                      echo "Removed $file"
```
Comment (tacaswell) [inline .github/workflows/tests.yml:343]:
```suggestion
                      echo "Removed ${base}.$extension"
```

?

Comment (Impaler343) [reply .github/workflows/tests.yml]: Fixed that. Another thing - Should I delete all empty folders inside the *result_images* folder? Or leave it as it is

Comment (Impaler343) [reply .github/workflows/tests.yml]: I guess this works well as it is, don't want to break anything in the process

**Review by tacaswell — APPROVED** [review]
The real test will be the next time we have broken test images and need to down load them...

**Comment by Impaler343** [comment]
Yup, do I need to rebase this?

**Comment by Impaler343** [comment]
This is probably ready to merge, do you expect any more changes from my side?

**Comment by rcomer** [comment]
@Impaler343 since @tacaswell already approved this one, you now need a second reviewer to also approve and then it can be merged (we require two approvals for most things except documentation changes).

**Comment by Impaler343** [comment]
ping @QuLogic, as he had reviewed a similar PR. Please review :)

**Review by ksunden — APPROVED** [review]

**Pull request merged**